File: Python/프로그래머스/Level2/튜플.py
import re

s = "{{20,111},{111}}"


def solution(s):
    answer = []
    s = s[2:-2]  # 문자열 슬라이싱
    s = s.split('},{')  # 리스트 형태로 반환
    s.sort(key=len)  # 문자열도 이런식으로 정렬이 가능

    for string in s:
        ar = string.split(',')
        print(ar)
        for el in ar:
            if int(el) not in answer:
                answer.append(int(el))
    print(answer)


solution(s)

File: Javascript/프로그래머스/Level2/튜플.js
function solution(s) {
    let newS = s // 숫자별로 자르는 작업, 길이에 따른 분류
        .substring(2, s.length - 2)
        .split('},{')
        .sort();

    let newArray = newS.map((el) => {
        // 배열을 구하는 작업
        return el.split(',');
    });
    newArray.sort((x, y) => x.length - y.length);
    var answer = [...newArray[0]];

    for (let i = 1; i < newArray.length; i++) {
        for (let el of answer) {
            newArray[i] = newArray[i].filter((x) => x !== el);
        }
        answer.push(...newArray[i]);
    }

    answer = answer.map((el) => parseInt(el)); // 원소들을 숫자로 바꾸는 과정
    console.log(answer);
    return answer;
}
